### PR TITLE
Unable to open empty remote workspace using vscode scheme URL

### LINKED ISSUE #223762: Unable to open empty remote workspace using `vscode` scheme URL
Does this issue occur when all extensions are disabled?: Yes - though you couldn't really open a remote URL without a plugin.

- VS Code Version: 1.91.1
- OS Version: Windows 11

When the main application is handling URL's with the `vscode` protocol, there are [two strategies](https://github.com/microsoft/vscode/blob/3b9aee0a96e6a812963418b08316c175e1386d16/src/vs/code/electron-main/app.ts#L904-L939) for determining the remote authority. If the URL contains an openable path, then [`getWindowOpenableFromProtocolUrl`](https://github.com/microsoft/vscode/blob/3b9aee0a96e6a812963418b08316c175e1386d16/src/vs/code/electron-main/app.ts#L805-L826) will transform a URL like `vscode://vscode-remote/wsl+ubuntu/mnt/c/GitDevelopment/monaco` to something like `vscode-remote://wsl+ubuntu/mnt/c/GitDevelopment/monaco`. If the URL does not contain a path after the authority (i.e. only one slash, like `vscode://vscode-remote/foo+bar`), then the remote authority is computed using the [`getRemoteAuthority`](https://github.com/microsoft/vscode/blob/3b9aee0a96e6a812963418b08316c175e1386d16/src/vs/platform/remote/common/remoteHosts.ts#L9-L11) function, which does _not_ handle URI's whose scheme is not already `vscode-remote`, so a `vscode://...` URL will always result in no `remoteAuthority`.

This may just be a question, if there's a way to construct a URL that will allow opening an empty remote workspace from e.g. a browser; otherwise, the logic may be buggy, and either it shouldn't delegate to `getRemoteAuthority` (because we don't expect the URL to have the right scheme), or `getRemoteAuthority` should be more flexible, something like this:

```typescript
export function getRemoteAuthority(uri: URI): string | undefined {
    if (uri.scheme === Schemas.vscodeRemote) { return uri.authority; }
    if (uri.authority === Schemas.vscodeRemote) {
        // ... extract the authority from the path, the same way `getWindowOpenableFromProtocolUrl` does it...
    }
    return undefined;
}
``` 

There is also a somewhat suspicious comment in the [`openFirstWindow`](https://github.com/microsoft/vscode/blob/3b9aee0a96e6a812963418b08316c175e1386d16/src/vs/code/electron-main/app.ts#L1256-L1279) method in the case of finding `initialProtocolUrls.urls` but not `initialProtocolUrls.openables`. In both cases, there is the comment `// remoteAuthority: will be determined based on openables`, but if only `urls` are found, then there are no openables, and so presumably any remote authorities associated with the `urls` are simply ignored, unless they're passed in some way I don't understand. Likely, the handler for the `initialProtocolUrls.urls` should also attempt to extract a remote authority from URL.

### PR DESCRIPTION
**Pull request by aeschli**
Fixes https://github.com/microsoft/vscode/issues/223762

### CODE REVIEW

**Review by bpasero — APPROVED** [review]

File: src/vs/code/electron-main/app.ts [hunk 1]
```diff
@@ -934,8 +936,7 @@ export class CodeApplication extends Disposable {
 				cli: { ...this.environmentMainService.args },
 				forceNewWindow: true,
 				forceEmpty: true,
-				gotoLineMode: true,
-				remoteAuthority: getRemoteAuthority(uri)
+				gotoLineMode: true
```
Comment (bpasero) [inline src/vs/code/electron-main/app.ts:939]: Intentional?

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/code/electron-main/app.ts (hunk 1 above)
Comment (aeschli) [reply src/vs/code/electron-main/app.ts:939]: I'm bringing it back.
I thought it was confusing/wrong that `getRemoteAuthority` is used on protocol URI, but then, getRemoteAuthority only returns something if the schema is `vscode-remote`.

**Review by bpasero — APPROVED** [review]

File: src/vs/code/electron-main/app.ts [hunk 2]
```diff
@@ -826,35 +826,37 @@ export class CodeApplication extends Disposable {
...
+			}
 
-				return { folderUri: remoteUri };
+			if (/:[\d]+$/.test(path)) {
+				// path with :line:column syntax
+				return { fileUri: remoteUri };
 			}
-		}
 
-		return undefined;
```
Comment (bpasero) [inline src/vs/code/electron-main/app.ts:857]: Removing this results in a warning

**Review by bpasero — CHANGES_REQUESTED** [review]

**Comment by aeschli** [comment]
Ok, now everything should work , Tested when running from sources: 
`.\scripts\code.bat --open-url vscode-oss://vscode-remote/test+test`
Thanks for reviewing, @bpasero, I was still in vacation mode...

File: src/vs/code/electron-main/app.ts [hunk 3]
```diff
@@ -826,34 +826,37 @@ export class CodeApplication extends Disposable {
 			//   To: vscode-remote://wsl+ubuntu/mnt/c/GitDevelopment/monaco
 
 			const secondSlash = uri.path.indexOf(posix.sep, 1 /* skip over the leading slash */);
+			let authority, path;
```
Comment (bpasero) [inline src/vs/code/electron-main/app.ts:829]: I dont want to be nit picking, but our code style is:

```ts
let authority: string;
let path: string;
```

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/code/electron-main/app.ts (hunk 3 above)
Comment (aeschli) [reply src/vs/code/electron-main/app.ts:829]: I wasn't aware of that, but yes, sure.

**Review by bpasero — APPROVED** [review]

**Pull request merged**